### PR TITLE
Rebase to latest upstream release

### PR DESCRIPTION
**Pull request by alexander-demichev**
This commit rebases the cluster-api-provider-azure openshift patches on top of the kubernetes-sigs/ cluster-api-provider-azure main branch after.
There are several commits that we carry on top of the upstream cluster-api-provider-azure and the rebase process allows us to preserve those. Here is a description of the process I used to create this PR.

(replicated @elmiko's process within [openshift/kubernetes-autoscaler](https://github.com/openshift/kubernetes-autoscaler/pull/164))

Process

First we need to identify the carry commits that we currently have, this is done against our previous rebase (or fork in this case) to catch new changes. Once identified we will drop commits which have merged upstream and only carry unique commits. (see below for the carried and dropped commits).

Identify carry commits:
```
git log --oneline --no-merges c356b7e776a70f5ce70a039641ef5b2e992d2d43..openshift/master
```

where b4be01 is the last upstream commit in our fork. This is the
list of commits we will need to apply onto the new upstream version of the
cluster-api. ideally, some of these commits can be dropped.

After identifying the carry commits, the next step is to create the new commit-tree that will be used for the rebase and then cherry pick the carry commits into the new branch. The following commands cover these steps:
```
$ git remote update # make sure we update our refs
$ git checkout c356b7e776a70f5ce70a039641ef5b2e992d2d43
$ git checkout -b merge-tmp # create a temporary branch for our merge commit
$ git checkout openshift/master # we want to be at the tip of the openshift master branch when we run the next command
$ echo 'merge latest upstream changes' | git commit-tree merge-tmp^{tree} -p HEAD -p merge-tmp -F - # create a new merge commit for our history
deadbeef12345678 # id of new merge commit
$ git branch rebase-upstream-latest deadbeef12345678 # create a new branch for the cherry-pick work
$ git checkout rebase-upstream-latest
$ git cherry-pick <carry commits> # cherry pick the needed commits into the new branch
```
With the rebase-upstream-latest branch in place, I cherry picked the carry commits which we should carry.

Carried Commits

These commits  are integral to our CI platform, or are specific to the releases we create for OpenShift.
```
   498c4ae9e UPSTREAM: <carry>: Add vendor directory
   3c05c04f3 UPSTREAM: <carry>: Add openshift specific changes
```

Changed Commits

```
---
```

Dropped Commits
<details>
  <summary>Click to expand</summary>

-   02b96338 UPSTREAM: <carry>: update golang version to 1.17
-   987eadd0 UPSTREAM: <carry>: add BUILD_IMAGES to Makefile and update nondocker builds
-   6cf23a67 UPSTREAM: <carry>: vendor: Bump API for Image Type field
-   796ab2b1 UPSTREAM: <carry>: Add VM Support for Marketplace Purchase Plans
-   2040aeac UPSTREAM: <carry>: Revert: Populate Marketplace Purchase Plan
-   06a8c54b UPSTREAM: <carry>: Populate Marketplace Purchase Plan
-   23f979cd UPSTREAM: <carry>: openshift: Migrate to opeshift/api
-   6d354dfb UPSTREAM: <carry>: openshift: Update dependecies
-   d1640552 UPSTREAM: <carry>: openshift: Add stack hub support to actuator
-   5cbccc03 UPSTREAM: <carry>: openshift: Set up stack hub client
-   a47741f3 UPSTREAM: <carry>: openshift: Add stack hub services
-   9ed7cbfb UPSTREAM: <carry>: openshift: Add availability set name field to api
-   047270d0 UPSTREAM: <carry>: openshift: Revendor azure sdk
-   c5901272 UPSTREAM: <carry>: Update machine-api-operator dependencies
-   6c62b91e UPSTREAM: <carry>: openshift: Fix AS spec pointer
-   d7b1ba91 UPSTREAM: <carry>: openshift: Skip AS creation when machineset label is missing
-   c57dbd02 UPSTREAM: <carry>: openshift: Add function for getting AS name
-   dbc5db2b UPSTREAM: <carry>: openshift: Add unit tests for availability sets
-   6c40684e UPSTREAM: <carry>: openshift: Create availability set if no zone found for region
-   ebf347ae UPSTREAM: <carry>: openshift: Add availabilitysets service
-   f58b315e UPSTREAM: <carry>: Enable accelerated networking
-   50df88b2 UPSTREAM: <carry>: Azure library updates to support accelerated networking
-   c689e78f UPSTREAM: <carry>: Ensure HOME is set for cache setup
-   ee2200e7 UPSTREAM: <carry>: Update dependencies to K8s 1.22
-   a7c40345 UPSTREAM: <carry>: Bug 1993159: set Machine Spec during update
-   1040454a UPSTREAM: <carry>: Bug 1988992: Create event only if the machine was modified
-   5d94c794 UPSTREAM: <carry>: openshift: Fix spot instances price type
-   7f78abea UPSTREAM: <carry>: openshift: Add in container vendor
-   6bed0c3c UPSTREAM: <carry>: openshift: Updating ose-azure-machine-controllers images to be consistent with ART Reconciling with https://github.com/openshift/ocp-build-data/tree/691e628254f318ce56efda5edc7448ec743c37b8/images/ose-azure-machine-controllers.yml
-   9dbeb200 UPSTREAM: <carry>: Update Git History check script to align with other repositories
-   83010763 UPSTREAM: <carry>: openshift: Make test names more descriptive, fix wrong named test.
-   3323fac2 UPSTREAM: <carry>: openshift: Improve logging message in case if vm not found
-   b7310cbb UPSTREAM: <carry>: openshift: Changed Exist method logic, now all ProvisioningStates treats as existent, only in case if object not found machine counts non existent. Failed and Deleting provisioning states return error, for future reconcile. Logging extended.
-   18bb9008 UPSTREAM: <carry>: openshift: Introduce syncPeriod for reconciliation
-   b0249058 UPSTREAM: <carry>: openshift: update controller-runtime dependency
-   570afa59 UPSTREAM: <carry>: openshift: support dual-stack interfaces
-   b2eda16d UPSTREAM: <carry>: openshift: Update mao dependency for webhook v1 changes
-   650c2cf3 UPSTREAM: <carry>: openshift: ensure CGO_ENABLED=0 on all makefile commands that need it
-   b70b9efc UPSTREAM: <carry>: openshift: update k8s dependencies to 1.21
-   987bc3d6 UPSTREAM: <carry>: openshift: Update makefile to be able to run with podman and docker,change scripts to run without contianer
-   eb94fe69 UPSTREAM: <carry>: openshift: Only log error on wrong instance type for scale from zero
-   053eaaec UPSTREAM: <carry>: openshift: Ensure original conditions aren't mutated during reconcile
-   231df73c UPSTREAM: <carry>: openshift: Add condition to show actuator exists condition on machine
-   2a02dfeb UPSTREAM: <carry>: openshift: Ensure response body is closed when we are finished with the request
-   adf818b8 UPSTREAM: <carry>: openshift: Updating ose-azure-machine-controllers builder & base images to be consistent with ART Reconciling with https://github.com/openshift/ocp-build-data/tree/5a1293dd0f380abf50c12d65c36655486d7745d0/images/ose-azure-machine-controllers.yml
-   3442c7a3 UPSTREAM: <carry>: openshift: add SecurityProfile.EncryptionAtHost parameter to enable host-based VM encryption
-   a4c0a838 UPSTREAM: <carry>: openshift: Updating Azure VMSize list from autoscaler.
-   c66f7ba3 UPSTREAM: <carry>: openshift: Ensure Machine is marked interruptible as well as Node
-   2caa1073 UPSTREAM: <carry>: openshift: Update dependency for golang.org/x/crypto
-   435b4dec UPSTREAM: <carry>: openshift: Fix controllers for updated controller-runtime
-   5c767009 UPSTREAM: <carry>: openshift: Bump K8s dependencies to 1.20
-   075372e2 UPSTREAM: <carry>: openshift: Ensure MAO dependency is up to date to include newest metrics
-   88b90230 UPSTREAM: <carry>: openshift: Update termination handler to use node conditions
-   608ffa06 UPSTREAM: <carry>: openshift: Bump scripts to Go 1.15
-   3b380c88 UPSTREAM: <carry>: openshift: Bump Azure autorest/adal to v0.9.5
-   d38af569 UPSTREAM: <carry>: openshift: bump Azure SDK version
-   45a93271 UPSTREAM: <carry>: openshift: Add verify-diff check in generate task
-   4090a697 UPSTREAM: <carry>: openshift: Adding managed disk parameters to support DES encryption
-   6c9c9593 UPSTREAM: <carry>: openshift: add bugzilla component and subcomponent to OWNERS
-   9b790fe4 UPSTREAM: <carry>: openshift: Fix missing logs due to mixed klog versions
-   04e680e5 UPSTREAM: <carry>: openshift: Fix missing logs due to mixed klog versions
-   dd5e7116 UPSTREAM: <carry>: openshift: Ensure the Virtual Machine provider state is set to Unknown when Failed
-   86841cbd UPSTREAM: <carry>: openshift: machine: Return full invalid configuration error
-   274ce1d5 UPSTREAM: <carry>: openshift: vm: Add AdditionalUnattendContent on Windows machines
-   cd15c1c5 UPSTREAM: <carry>: openshift: Bug 1878880: re-vendor machine-api-operator at e0db6b65
-   5862e3d3 UPSTREAM: <carry>: openshift: Bump K8s dependencies to v1.19.0
-   db990d84 UPSTREAM: <carry>: openshift: Updating Dockerfile.rhel baseimages to match ocp-build-data config
-   72763932 UPSTREAM: <carry>: openshift: Fail machine if create request doesn't succeed
-   72169c58 UPSTREAM: <carry>: openshift: Add Dockerfile.rhel to match build configuration in ocp-build-data
-   29efd367 UPSTREAM: <carry>: openshift: update make generate
-   201dd37f UPSTREAM: <carry>: openshift: Slow the default lease retry and renew rate for machine controller
-   f138f6bb UPSTREAM: <carry>: openshift: Assert VM exists if VM state is Creating
-   0d53a0d9 UPSTREAM: <carry>: openshift: Ensure Interruptible Instance Label is propogated onto the Node
-   5955dafd UPSTREAM: <carry>: openshift: Add unit tests for different cloud envs
-   5279a031 UPSTREAM: <carry>: openshift: Update dependencies
-   d95ffcff UPSTREAM: <carry>: openshift: Add support for different cloud environments
-   3248f7db UPSTREAM: <carry>: openshift: Increase leader election lease time
-   f1d6f90d UPSTREAM: <carry>: openshift: Revendor mao to bring https://github.com/openshift/machine-api-operator/pull/644
-   3dad165e UPSTREAM: <carry>: openshift: Ensure empty max price does not cause error when creating a spot VM
-   ab01fdec UPSTREAM: <carry>: openshift: Make dockerfile consistent with other providers
-   044a01d7 UPSTREAM: <carry>: openshift: Set spot label in machine spec
-   85d1eca9 UPSTREAM: <carry>: openshift: Add termination handler tests
-   d1ac2e14 UPSTREAM: <carry>: openshift: Add termination handler
-   3cda328d  UPSTREAM: <carry>: openshift: Add termination handler to Makefile and Dockerfile
-   88a538be UPSTREAM: <carry>: openshift: Bump golang.org/x/text to v0.3.3
-   fa840d10 UPSTREAM: <carry>: openshift: Revendor MAO
-   2e64806c UPSTREAM: <carry>: openshift: Add metrics for machine controllers
-   d46153e4 UPSTREAM: <carry>: openshift: Skip SSH setup on Windows VM
-   7fe5391f UPSTREAM: <carry>: openshift: Implement leader election for azure provider
-   b85933ff UPSTREAM: <carry>: openshift: Update azure sdk
-   7d9c9095 UPSTREAM: <carry>: openshift: Add support for spot instances
-   ff906630 UPSTREAM: <carry>: openshift: Fix confusing log in azure
-   6183e484 UPSTREAM: <carry>: openshift: Run make generate
-   b7e45ab1 UPSTREAM: <carry>: openshift: add make generate to generate deepcopy code
-   58e4ac0f UPSTREAM: <carry>: openshift: vendor k8s.io/code-generator
-   f74f13d5 UPSTREAM: <carry>: openshift: Add configv1 to scheme
-   ddd15dfb UPSTREAM: <carry>: openshift: Revendor MAO
-   5d5d67e2 UPSTREAM: <carry>: openshift: Expose spot instances to provider spec and reconcile label
-   e66168e9 UPSTREAM: <carry>: openshift: Rename machine controller import alias
-   72e2c4c1 UPSTREAM: <carry>: openshift: Add health checks
-   af08a66b UPSTREAM: <carry>: openshift: Add dgrigore to approvers
-   17d4edc5 UPSTREAM: <carry>: openshift: Update machine-api-operator-dependency
-   5840552c UPSTREAM: <carry>: openshift: Use ToLower for Subscription ID and Resource Group Name in Machine ProviderID
-   325e1d5a UPSTREAM: <carry>: openshift:  Add unknown state for externally removed machines
-   cc1374df UPSTREAM: <carry>: openshift: Bump all dependencies to kube 1.18
-   9034812d UPSTREAM: <carry>: openshift: Removed deprecated v1apha1 api
-   639c0654 UPSTREAM: <carry>: openshift: Update dependencies
-   ac0a2540 UPSTREAM: <carry>: openshift: Return invalid configuration errors when creating NICs with invalid config
-   cdc42ef6 UPSTREAM: <carry>: openshift: Fix breaks due to updated dependencies
-   54c676aa UPSTREAM: <carry>: openshift: Update dependencies
-   8e312518 UPSTREAM: <carry>: openshift: Use Go 1.13 errors instead of github.com/pkg/errors
-   ad6ae030 UPSTREAM: <carry>: openshift: Update vendor
-   98d5d87a UPSTREAM: <carry>: openshift: Remove MachineSet manager from image
-   73a28c17 UPSTREAM: <carry>: openshift: Update golang/crypto to bring CVE-2020-9283 fix
-   7bdbd729 UPSTREAM: <carry>: openshift: Bump to Go 1.13
-   3fd3692f UPSTREAM: <carry>: openshift: Update OWNERS with current team members
-   0dc1da09 UPSTREAM: <carry>: openshift: Restore original make unit behaviour
-   a3d66b63 UPSTREAM: <carry>: openshift: Remove bison from OWNERS file
-   4fc8bbeb UPSTREAM: <carry>: openshift: Support private IPv6 configurations
-   b304c7bf UPSTREAM: <carry>: openshift: Revendor mao with timeout formatting fix
-   a0b4361c UPSTREAM: <carry>: openshift: Don't re-reconcile on invalid configuration
-   d015d541 UPSTREAM: <carry>: openshift: Add envtest binaries for testing
-   9db40c29 UPSTREAM: <carry>: openshift: make vendor
-   85de3bae UPSTREAM: <carry>: openshift: Add envTest tests for reconciler
-   07a1c297 UPSTREAM: <carry>: openshift: Add tests for reconcile
-   d3921c6a UPSTREAM: <carry>: openshift: Add machineset controller
-   8b91ed17 UPSTREAM: <carry>: openshift: Remove fqdn from public ip creation
-   d80999fe UPSTREAM: <carry>: openshift: add docker and build task to makefile
-   e441cdfb Add bin/ directory to gitignore
-   72c8756a UPSTREAM: <carry>: openshift: Remove modification of LastUpdate in machine scope
-   832a43a3 UPSTREAM: <carry>: openshift: Remove deprecated check in update
-   8bfc93ea UPSTREAM: <carry>: openshift: Revendor MAO with new node draining feature
-   06be56f2 UPSTREAM: <carry>: openshift: Save status before Patch()
-   8a9592f1 UPSTREAM: <carry>: openshift: OWNERS: remove frobware
-   611a78a4 UPSTREAM: <carry>: openshift: Add unit test task to makefile
-   7ec5691b UPSTREAM: <carry>: openshift: Replace update with patch
-   937b4e62 UPSTREAM: <carry>: openshift: Vendor dependencies
-   52c22326 UPSTREAM: <carry>: openshift: Remove unnecessary machine specific client
-   f3f269f1 UPSTREAM: <carry>: openshift: Update owners
-   d9d24c57 UPSTREAM: <carry>: openshift: allow managed identity to be unset
-   925b15bc UPSTREAM: <carry>: openshift: Bring mao and drop cluster struct references to satisfy actuator interface
-   c3ff86a9 UPSTREAM: <carry>: openshift: Revendor to drop openshift/cluster-api in favour of machine-api-operator
-   39f21d65 UPSTREAM: <carry>: openshift: Fix statusCode conversion
-   d6bb862d UPSTREAM: <carry>: openshift: update scripts for enabling go modules
-   9f6020e9 UPSTREAM: <carry>: openshift: revendor with go modules
-   5f5e557b UPSTREAM: <carry>: openshift: remove unused make targets
-   4b17bb4e UPSTREAM: <carry>: openshift: delete unused files
-   cbe6bdd8 UPSTREAM: <carry>: openshift: Fail with InvalidMachineConfiguration in Create when appropriate
-   c1d771c6 UPSTREAM: <carry>: openshift: services/networkinterfaces: fix application secuirty group collection
-   593cfd46 UPSTREAM: <carry>: openshift: api: allow setting the security group and application security groups for machine in spec
-   d1d05501 UPSTREAM: <carry>: openshift: services/newtorkinterfaces: allow providing the network security group and application security groups for the interface.
-   21aa5c66 UPSTREAM: <carry>: openshift: services: add service to handle application security groups
-   08ab4331 UPSTREAM: <carry>: openshift: Fix bad references to cluster/v1alpha1
-   3e3c9dc9 UPSTREAM: <carry>: openshift: Drop unused certificates service to drop kubeadmin deps
-   57039735 UPSTREAM: <carry>: openshift: Revendor for kube-1.16.0
-   719f40a4 UPSTREAM: <carry>: openshift: Revendor to bring https://github.com/openshift/cluster-api/pull/72
-   32945951 UPSTREAM: <carry>: openshift: Remove dead code
-   35cc61a8 UPSTREAM: <carry>: openshift: tag instances with cluster id
-   3f273892 UPSTREAM: <carry>: openshift: Resync every 20 seconds
-   432573f8 UPSTREAM: <carry>: openshift: Apply 'make goimports'
-   3902dd77 UPSTREAM: <carry>: openshift: Revendor to bring cluster-api bd4f0ba460758c33c609cb333721606c6d002c4e
-   b066ec0b UPSTREAM: <carry>: openshift: Update machine only when changed
-   e47ba9be UPSTREAM: <carry>: openshift: Skip deletion of public IPs with too long names
-   ad61b3d0 UPSTREAM: <carry>: openshift: Run 'make goimports'
-   286810e3 UPSTREAM: <carry>: openshift: Extend makefile with 'make goimports' target
-   309b4c23 UPSTREAM: <carry>: openshift: do not randomize public ip name, error instead
-   a23858d8 UPSTREAM: <carry>: openshift: Check if annotations are initialized before assigning a key
-   e90ca70b UPSTREAM: <carry>: openshift: Drop ssh SSHPrivateKey
-   408d4c81 UPSTREAM: <carry>: openshift: Add unit test for exists()
-   e93128fe UPSTREAM: <carry>: openshift: update deps and bazel generated files
-   ee1b0701 UPSTREAM: <carry>: openshift: generate public ip name with at most 63 characters
-   85c2f992 UPSTREAM: <carry>: openshift: create git history verification script
-   44ea1c00 UPSTREAM: <carry>: openshift: Add support for machine tags
-   ed6726c6 UPSTREAM: <carry>: openshift: use location from API over secret and unit test
-   8000a3e2 UPSTREAM: <carry>: openshift: run 'dep ensure' to bring new deps
-   74789480 UPSTREAM: <carry>: openshift: set additional machine annotations/labels with instance bits
-   b05edbde UPSTREAM: <carry>: openshift: Remove mutating logic from exist and set vm state in update
-   4c00262d UPSTREAM: <carry>: openshift: Rename close to persist, make it error aware, stop persisting on exist
-   42ca4557 UPSTREAM: <carry>: openshift: Add omitempty tag so empty strings are omitted from the output
-   17a94153 UPSTREAM: <carry>: openshift: Makefile: add e2e machinations
-   7edc0cb8 UPSTREAM: <carry>: openshift: Remove dep on bazel binary when running test-e2e target
-   6a50a8c5 UPSTREAM: <carry>: openshift: Move machineSpec from azureprovider.k8s.io/v1alpha1 into azureproviderconfig.openshift.io/v1beta1
-   dafd985a UPSTREAM: <carry>: openshift: Allow to run disk delete ops async
-   36124f1d UPSTREAM: <carry>: openshift: print successfully deleted vm only when err is nil
-   b0bf44fb Verify bazel
-   303eaa25 UPSTREAM: <carry>: openshift: Stop asigning random az
-   11996447 UPSTREAM: <carry>: openshift: services/virtualnetworks: create in NetworkResourceGroup
-   be195bc2 UPSTREAM: <carry>: openshift: services/subnets: create in NetworkResourceGroup
-   4e1e8a1f UPSTREAM: <carry>: openshift: Allow for separate Network resource group
-   cd9c9573 UPSTREAM: <carry>: openshift: Update vendored github.com/openshift/kubernetes-drain
-   28517a02 UPSTREAM: <carry>: openshift: run 'dep ensure' to bring new deps
-   cbf0effe UPSTREAM: <carry>: openshift: Record events for actuator CUD operations
-   81fbb766 UPSTREAM: <carry>: openshift: run 'make generate'
-   47833918 UPSTREAM: <carry>: openshift: allowing choosing availability zone for a machine
-   04051e44 UPSTREAM: <carry>: openshift: Expect vnet and managed identity to be always set
-   f1d7e012 UPSTREAM: <carry>: openshift: Switch builds to use Go 1.12
-   f2a9dca3 UPSTREAM: <carry>: openshift: describe how to deploy machine API with azure machine controller over kubernetes
-   790f3ac0 UPSTREAM: <carry>: openshift: Deploy machine API plane with azure actuator
-   7f07df63 UPSTREAM: <carry>: openshift: Provide a list of RBAC rules for the actuator
-   12d3c939 UPSTREAM: <carry>: openshift: Allow vnet and managed identity names to be set explicitly
-   c0d30087 UPSTREAM: <carry>: openshift: Add namespace flag
-   0f6d8d2d UPSTREAM: <carry>: openshift: Update code to accept k8s 1.14
-   72dedbce UPSTREAM: <carry>: openshift: sync openshift/cluster-api@openshift-4.2-cluster-api-0.1.0 and k8s to 1.14.1
-   702ef3a4 UPSTREAM: <carry>: openshift: Set instance subnet and load balancers explicitly
-   43af035c UPSTREAM: <carry>: openshift: Allow to run Create/Delete ops async
-   7ae13d77 UPSTREAM: <carry>: openshift: Set internal DNS name
-   ede8126b UPSTREAM: <carry>: openshift: Publish machine conditions during Create and Update operations
-   adf47822 UPSTREAM: <carry>: openshift: Set machine.Spec.ProviderID
-   77d9b204 UPSTREAM: <carry>: openshift: Set machine addresses
-   4ee3fce1 UPSTREAM: <carry>: openshift: Allow to set public IP address when creating vm
-   4986149e UPSTREAM: 202: openshift: delete OS Disk and network interface before creating new VM instance, else fails due to previous existence
-   a2707106 UPSTREAM: <carry>: openshift: Update README.md
-   84af9477 UPSTREAM: <carry>: openshift: remove kubeadm cluser configuration from clusterprovider
-   7e23f237 UPSTREAM: <carry>: openshift: build fixes due to new cluster-api vendor and addition of managed identity
-   fa79861a UPSTREAM: <carry>: openshift: revendor cluster-api dependency
-   4f7348df UPSTREAM: <carry>: openshift: convert machine label into constant for easier change in future
-   cdf874c4 UPSTREAM: <carry>: openshift: revendor after previous changes
-   92a7c160 UPSTREAM: <carry>: openshift: Build both openshift cluster-api manager and azure-provider manager
-   34a3a8cd UPSTREAM: <carry>: openshift: revendor after previous changes
-   9856a74d UPSTREAM: <carry>: openshift: Use openshift cluster-api branch instead of sigs.k8s.io
-   9a42fd6e UPSTREAM: <carry>: openshift: revendor after previous changes
-   01eca66c UPSTREAM: <carry>: openshift: decouple machine actuator from cluster object
-   de271dd9 UPSTREAM: <carry>: openshift: support for resourceid as an alternate image location
-   bb8dae8b UPSTREAM: <carry>: openshift: simplify if ... else logic
-   260e7c78 UPSTREAM: <carry>: openshift: revendor after previous changes
-   4755563f UPSTREAM: <carry>: openshift: Add support for Custom data, read from Secrets for Ignition
-   5c0e9c57 fix: Support Availability Zones based on regions, automatically known using ARM API (#155)
-   3d72a4eb fix: Generate consistent names for Azure resources (#149)
-   14a13e81 Upstream: <carry>: openshift: Add docker images for CI
-   33268448 fix: Restructure machine actuator for unittests, and add unittests (#145)
-   bdbc5bd7 Remove hardcoded values from virtual machines, respect values from machineconfig instead (#147)
-   9a6f513f Update dependencies (#143)
-   a818b2d9 fix: New Compute and Network micro services convering machine, cluster to pure ARM calls (#139)
-   803aefe2 fix: Create a new Generic Service interface which provides an individual azure service easy for unittesting (#138)
-   cde05d6a Create certificates, admin kubeconfig, bootstrap token and discoveryhashes using kubeadm (#136)
-   fc16fa80 UPSTREAM: <carry>: openshift: Update owners
</details>

### CODE REVIEW

File: OWNERS [hunk 1]
```diff
@@ -1,20 +1,10 @@
...
-
-labels:
-  - sig/cluster-lifecycle
-  - area/provider/azure
+  - damdo
+  - alexander-demichev
+  - JoelSpeed
+  - elmiko
+  - Fedosin
+  - lobziik
```
Comment (JoelSpeed) [inline OWNERS:10]: Probably not worth adding Denis here as he will be leaving the team shortly

File: openshift/Dockerfile.openshift [hunk 2]
```diff
@@ -0,0 +1,18 @@
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.17-openshift-4.10 AS builder
```
Comment (JoelSpeed) [inline openshift/Dockerfile.openshift:1]: Does this need to be 4.11 now?

**Review by Fedosin — APPROVED** [review]
/lgtm

**Pull request merged**